mailbox/store/src/main/java/org/apache/james/mailbox/store/event/MailboxUpdateParentAsLookupListener.java

```diff
@@ -0,0 +1,87 @@
+package org.apache.james.mailbox.store.event;
+
+import java.util.List;
+
+import org.apache.james.mailbox.MailboxListener;
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.exception.MailboxException;
+import org.apache.james.mailbox.model.MailboxACL;
+import org.apache.james.mailbox.model.MailboxACL.EntryKey;
+import org.apache.james.mailbox.model.MailboxPath;
+import org.apache.james.mailbox.store.MailboxSessionMapperFactory;
+import org.apache.james.mailbox.store.mail.MailboxMapper;
+import org.apache.james.mailbox.store.mail.model.Mailbox;
+import org.apache.james.mailbox.store.transaction.Mapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MailboxUpdateParentAsLookupListener implements MailboxListener {
+    private static final Logger logger = LoggerFactory.getLogger(MailboxUpdateParentAsLookupListener.class);
+    private MailboxSessionMapperFactory mailboxSessionMapperFactory;
+
+    public MailboxUpdateParentAsLookupListener(MailboxSessionMapperFactory mailboxSessionMapperFactory) {
+        this.mailboxSessionMapperFactory = mailboxSessionMapperFactory;
+    }
+    @Override
+    public ListenerType getType() {
+        return ListenerType.EACH_NODE;
+    }
+
+    @Override
+    public ExecutionMode getExecutionMode() {
+        return ExecutionMode.SYNCHRONOUS;
+    }
+
+    @Override
+    public void event(Event event) {
+        if (event instanceof EventFactory.MailboxLookupRightUpdate) {
+            try {
+                EventFactory.MailboxLookupRightUpdate mailboxLookupRightUpdate = (EventFactory.MailboxLookupRightUpdate)event;
+
+                MailboxSession mailboxSession = mailboxLookupRightUpdate.getSession();
+                MailboxPath mailboxPath = mailboxLookupRightUpdate.getMailboxPath();
+
+                MailboxMapper mapper = mailboxSessionMapperFactory.getMailboxMapper(mailboxSession);
+                List<MailboxPath> mailboxPaths = mailboxPath.getHierarchyLevels(mailboxSession.getPathDelimiter());
+                mailboxPaths.stream()
+                    .filter(hierarchyMailboxPath -> !hierarchyMailboxPath.equals(mailboxPath))
+                    .forEach(parentMailboxPath -> {
+                        updateLookupRight(mapper, parentMailboxPath, mailboxLookupRightUpdate.getEntryKey());
+                    });
+            } catch (MailboxException e) {
+                logger.error("Unable to look up MailboxMapper", e);
+            }
+        }
+    }
+
+    private void updateLookupRight(MailboxMapper mapper, MailboxPath mailboxPath1, EntryKey entryKey) {
+        try {
+            Mailbox mailbox = mapper.findMailboxByPath(mailboxPath1);
+            MailboxACL.ACLCommand mailboxACLCommand = MailboxACL.command()
+                .key(entryKey)
+                .rights(MailboxACL.Right.Lookup)
+                .asAddition();
+            mapper.execute(Mapper.toTransaction(() -> mapper.updateACL(mailbox, mailboxACLCommand)));
```

Comment:
Mmmmm.... I'm upset... I would like to have ACLEvents for these updates!

Of course, we should avoid reprocessing the events that we are generating here (even if this is an optimization, and would have no correctness effects), but we might imagine subsystems (IE JMAP pubSub?) that relies on these events, **and we can not afford not telling them what happens**. I smell architectural level hard to debug bugs coming from this.